library/public/types.h

```diff
@@ -210,6 +200,11 @@ class direction_t : public double_array_t<3>
   {
     return (*this)[2];
   }
+  [[nodiscard]] operator f3d::vector3_t() const
+  {
+    // XXX: No better way to do this ?
+    return f3d::vector3_t{ this->x(), this->y(), this->z() };
```

Comment:
@Meakk any better ideas ?

Reply:
`return f3d::vector3_t{ this->Array };`? Requires `Array` in `protected` rather than `private`

Reply:
right